core-rust/mesh-api-server/src/mesh_api/handlers/construction_parse.rs

```diff
@@ -0,0 +1,197 @@
+use crate::prelude::*;
+use models::AccountIdentifier;
+use models::{Operation, OperationIdentifier};
+use radix_engine_interface::blueprints::account::{
+    AccountTryDepositOrAbortManifestInput, AccountWithdrawManifestInput,
+};
+use radix_transactions::manifest::{CallMethod, TakeFromWorktop};
+use radix_transactions::validation::TransactionValidator;
+
+pub(crate) async fn handle_construction_parse(
+    state: State<MeshApiState>,
+    Json(request): Json<models::ConstructionParseRequest>,
+) -> Result<Json<models::ConstructionParseResponse>, ResponseError> {
+    assert_matching_network(&request.network_identifier, &state.network)?;
+
+    let transaction_bytes = hex::decode(&request.transaction).map_err(|_| {
+        ResponseError::from(ApiError::InvalidTransaction)
+            .with_details(format!("Invalid transaction hex: {}", &request.transaction))
+    })?;
+    let (instructions, signers) = if request.signed {
+        let transaction =
+            NotarizedTransactionV1::from_raw(&RawNotarizedTransaction::from_vec(transaction_bytes))
+                .map_err(|_| {
+                    ResponseError::from(ApiError::InvalidTransaction)
+                        .with_details(format!("Invalid transaction: {}", &request.transaction))
+                })?;
+        let validated = transaction
```

Comment:
We'd then need to match on the version to extract the instructions, but can then have the `parse_instructions` method take an `impl Iterator<Item = AnyInstruction>`, and call `instructions.iter().map(|i| i.clone().into_any())` for each version in the match.

Reply:
EDIT: Given that the instruction interpretation is very tight and only accepts transactions constructed with the Mesh API, I'm happy to leave this supporting just those transactions.

I'd suggest some code comment somewhere explaining that though, to prevent someone picking this up later from being confused.

Reply:
Comment added